### PR TITLE
PMM-2530: Add timeout flags

### PR DESCRIPTION
**Pull request by unguiculus**
Adds the following flags:

* -mongodb.socket-timeout
* -mongodb.sync-timeout

I left the defaults as they were, but I think it might be a good idea to change them as well. The MongoDB driver has a default for the sync timeout of 7 seconds ([session.go#L1706](https://github.com/percona/mongodb_exporter/blob/develop/vendor/gopkg.in/mgo.v2/session.go#L1706)) and a default value for the socket timeout of one minute ([session.go#L1716](https://github.com/percona/mongodb_exporter/blob/develop/vendor/gopkg.in/mgo.v2/session.go#L1716)) whereas the exporter uses a much longer default for the sync timeout of one minute and an extremely short default of 3 seconds for the socket timeout.

We've found that the default socket timeout is way too short when you have a recovering replica. It may take quite a bit to connect. You want to make sure to get your metrics, so I'd rather set the timeout higher.

### CODE REVIEW

**Comment by unguiculus** [comment]
Any chance this gets merged?

File: mongodb_exporter.go [hunk 1]
```diff
@@ -62,6 +63,9 @@ var (
 	maxConnectionsF               = flag.Int("mongodb.max-connections", 1, "Max number of pooled connections to the database.")
 	testF                         = flag.Bool("test", false, "Check MongoDB connection, print buildInfo() information and exit.")
 
+	socketTimeoutF = flag.Int64("mongodb.socket-timeout", int64(3*time.Second), "Amount of time in ms to wait for a non-responding socket to the database before it is forcefully closed")
+	syncTimeoutF   = flag.Int64("mongodb.sync-timeout", int64(1*time.Minute), "Amount of time in ms an operation with this session will wait before returning an error in case a connection to a usable server can't be established")
```
Comment (arvenil) [inline mongodb_exporter.go]: There is [flag.Duration](https://golang.org/pkg/flag/#Duration) for this purpose.
Example https://github.com/percona/mysqld_exporter/blob/50377ef0863147751e3057fa95b263ed396ad2f4/mysqld_exporter.go#L82

**Comment by timvaillancourt** [comment]
@unguiculus Thanks for the contribution. Approved by me

File: mongodb_exporter.go (hunk 1 above)
Comment (timvaillancourt) [reply mongodb_exporter.go]: Good idea!

**Comment by unguiculus** [comment]
@arvenil @timvaillancourt Changed it to `flag.Duration`. Thanks.

**Pull request merged**